Sources/Templates/Extensions/Array+Extension.swift

```diff
@@ -0,0 +1,18 @@
+import Foundation
+
+public extension Array where Element: Hashable {
+
+    var distinct: Array {
+        var buffer = Array()
+        var added = Set<Element>()
+        
+        for element in self {
```

Comment:
https://www.avanderlee.com/swift/unique-values-removing-duplicates-array/ this is also a nice solution I think
```swift
extension Sequence where Iterator.Element: Hashable {
    func unique() -> [Iterator.Element] {
        var seen: Set<Iterator.Element> = []
        return filter { seen.insert($0).inserted }
    }
}
```